### ISSUE TITLE
[Enhancement]: Resource block - aws_kinesis_firehose_delivery_stream - destination option should include "opensearch" along with "elasticsearch"

### ISSUE THREAD

**Issue by anithapriyanatarajan** [hashicorp/terraform-provider-aws#29091]
### Description

Currently, we are using destination as "elasticsearch" for both elasticsearch as well as opensearch. This is misleading and the terminologies are not matching the actual AWS service. Is there any roadmap for updating this or else please plan to include this?

### Affected Resource(s) and/or Data Source(s)

aws_kinesis_firehose_delivery_stream

### Potential Terraform Configuration

```terraform
resource "aws_kinesis_firehose_delivery_stream" "test" {

  name        = "terraform-kinesis-firehose-os"
  destination = "opensearch"
  s3_configuration {
    role_arn   = aws_iam_role.firehose.arn
    bucket_arn = aws_s3_bucket.bucket.arn
  }
  opensearch_configuration {
    domain_arn = aws_opensearch_domain.test_cluster.arn
    role_arn   = aws_iam_role.firehose.arn
    index_name = "test"
    type_name  = "test"

    vpc_config {
      subnet_ids         = [aws_subnet.first.id, aws_subnet.second.id]
      security_group_ids = [aws_security_group.first.id]
      role_arn           = aws_iam_role.firehose.arn
    }
  }
}
```

### Would you like to implement a fix?

None

**Comment by jamiejackson** [comment]
This bit me today and it led to more work than should have been necessary.

TL;DR: Importing an existing opensearch stream is arduous and requires a change from an opensearch destination to an elasticsearch destination.

Problem case:

- Create a Firehose delivery from WAF, to target AWS OpenSearch.
- Terraform import the aws_kinesis_firehose_delivery_stream resource
- The import ignores (probably doesn't undertand) the `AmazonopensearchserviceDestinationDescription` destination and instead uses (defaults to?) destination = `extended_s3`.
- The imported resource state has a completely different configuration profile than an opensearch (or elasticsearch) destination would, so the plan is fairly useless. Worse, there is now a destination type mismatch:
   ```
      ~ destination    = "extended_s3" -> "elasticsearch" # forces replacement
   ```
- The only way I knew how to move forward was to
  - save down the output of `aws firehose describe-delivery-stream --delivery-stream-name my_delivery_stream`
  - run the apply, which instead of changing from "extended_s3" to "elasticsearch" is really replacing the opensearch destination to an elasticsearch destination
  - Keep comparing the new `aws firehose describe-delivery-stream --delivery-stream-name my_delivery_stream` with the original until there were no diffs

This was not easy to work out. I'm also not sure if there will be any side effects of having an elasticsearch destination instead of an opensearch one.

Notes:

Opensearch aws firehose description:

```json
{
    "DeliveryStreamDescription": {
        "Destinations": [
            {
                ...
                "AmazonopensearchserviceDestinationDescription": {
                ...
                }
            }
        ],
        ...
    }
}
```

Elasticsearch version:

```json
{
    "DeliveryStreamDescription": {
        "Destinations": [
            {
                ...
                "ElasticsearchDestinationDescription": {
                ...
                }
            }
        ],
        ...
    }
}
```
